test/async_progress_worker.js

```diff
@@ -4,12 +4,146 @@ const common = require('./common');
 const assert = require('assert');
 
 module.exports = common.runTest(test);
+const nodeVersion = process.versions.node.split('.')[0];
+
+let asyncHooks;
+function checkAsyncHooks () {
+  if (nodeVersion >= 8) {
+    if (asyncHooks === undefined) {
+      asyncHooks = require('async_hooks');
+    }
+    return true;
+  }
+  return false;
+}
```

Comment:
According to the async hooks [doc](https://nodejs.org/api/async_hooks.html) their use is discouraged. Can we use the alternatives mentioned there?

Reply:
Hmm.. From what I understand, we use `async_hooks` here to verify that async resources/objects passed from the C++ side are successfully instantiated, and can be used to track the life cycle of the async call from JS. 

I took a look at `AsyncLocalStorage`, and it doesn't seem to provide any API to inspect the async resources/objects associated with an async call. @legendecas Any ideas?

Reply:
OK. That's fair.